client-core/src/wallet.rs

```diff
@@ -96,6 +96,7 @@ pub trait WalletClient: Send + Sync {
         name: &str,
         passphrase: &SecUtf8,
         wallet_kind: WalletKind,
+        mnemonics_word_count: u32,
```

Comment:
probably OK in this PR, but this argument doesn't make sense WalletKind::Basic, so perhaps a TODO comment for some API redesign/refactoring (e.g. as a part of this https://github.com/crypto-com/chain/issues/1491 ).
A bit cleaner way may be either to have separate functions `new_wallet_basic(...)` and `new_wallet_hd(...,  mnemonics_word_count: ...)` or e.g. put the parameter inside `WalletKind` enum's HD variant

Reply:
i'll add TODO